plugins/out_stackdriver/stackdriver.c

```diff
@@ -361,152 +361,172 @@ static struct mk_list *parse_local_resource_id_to_list(char *str, char *type)
-                }
-                else if (!ctx->pod_name) {
-                    ctx->pod_name = flb_sds_create(ptr->val);
-                }
-                else if (!ctx->container_name) {
-                    ctx->container_name = flb_sds_create(ptr->val);
+    if (!ctx->local_resource_id) {
+        flb_plg_error(ctx->ins, "local_resource_is is not assigned");
+        return -1;
+    }
+
+    len_k8s_container = sizeof(K8S_CONTAINER) - 1;
+    len_k8s_node = sizeof(K8S_NODE) - 1;
+    len_k8s_pod = sizeof(K8S_POD) - 1;
+
+    if (strncmp(type, K8S_CONTAINER, len_k8s_container) == 0) {
+        list = parse_local_resource_id_to_list(ctx->local_resource_id, K8S_CONTAINER);
+        if (!list) {
+            goto error;
+        }
+
+        /* iterate through the list */
+        mk_list_foreach_safe(head, tmp, list) {
+            ptr = mk_list_entry(head, struct local_resource_id_list, _head);
+            if (first) {
+                /* check the prefix */
+                if (flb_sds_len(ptr->val) != len_k8s_container ||
+                    strncmp(ptr->val, K8S_CONTAINER, len_k8s_container) != 0) {
+                    goto error;
                 }
+                first = 0;
+                continue;
             }
 
-            if (!ctx->namespace_name || !ctx->pod_name || !ctx->container_name) {
-                goto error;
+            /* Follow the order of fields in local_resource_id */
+            if (!ctx->namespace_name) {
+                ctx->namespace_name = flb_sds_create(ptr->val);
             }
-        }
-        else if (strncmp(type, K8S_NODE, len_k8s_node) == 0) {
-            list = parse_local_resource_id_to_list(local_resource_id, K8S_NODE);
-            if (!list) {
-                goto error;
+            else if (!ctx->pod_name) {
+                ctx->pod_name = flb_sds_create(ptr->val);
             }
+            else if (!ctx->container_name) {
+                ctx->container_name = flb_sds_create(ptr->val);
+            }
+        }
 
-            mk_list_foreach_safe(head, tmp, list) {
-                ptr = mk_list_entry(head, struct local_resource_id_list, _head);
-                if (first) {
-                    if (flb_sds_len(ptr->val) != len_k8s_node ||
-                        strncmp(ptr->val, K8S_NODE, len_k8s_node) != 0) {
-                        goto error;
-                    }
-                    first = 0;
-                    continue;
-                }
+        if (!ctx->namespace_name || !ctx->pod_name || !ctx->container_name) {
+            goto error;
+        }
+    }
+    else if (strncmp(type, K8S_NODE, len_k8s_node) == 0) {
+        list = parse_local_resource_id_to_list(ctx->local_resource_id, K8S_NODE);
+        if (!list) {
+            goto error;
+        }
 
-                if (ptr != NULL) {
-                    ctx->node_name = flb_sds_create(ptr->val);
+        mk_list_foreach_safe(head, tmp, list) {
```

Comment:
```mk_list_foreach_safe()``` is primary used when one of the nodes of the list while iterating might be deleted. If is just a read-only operation you have to use ```mk_list_foreach()```

Reply:
updated